packages/gafl-webapp-service/src/pages/terms-and-conditions/__tests__/route.spec.js

```diff
@@ -13,4 +11,108 @@ describe('terms-and-conditions > route', () => {
       expect(pageRoute).toBeCalledWith('terms-and-conditions', '/buy/conditions', validator, nextPage, getData)
     })
   })
+
+  describe('getData', () => {
+    const mockRequest = (statusGet = () => {}, transactionGet = () => {}) => ({
+      cache: () => ({
+        helpers: {
+          transaction: {
+            get: transactionGet
+          },
+          status: {
+            getCurrentPermission: statusGet
+          }
+        }
+      })
+    })
+
+    beforeEach(() => jest.clearAllMocks())
+
+    it('LICENCE_SUMMARY redirect', async () => {
+      const status = () => ({
+        [LICENCE_SUMMARY.page]: false
+      })
+      const func = () => getData(mockRequest(status))
+      await expect(func).rejects.toThrowRedirectTo(LICENCE_SUMMARY.uri)
+    })
+
+    it('CONTACT_SUMMARY redirect', async () => {
+      const status = () => ({
+        [LICENCE_SUMMARY.page]: true,
+        [CONTACT_SUMMARY.page]: false
+      })
+      const func = () => getData(mockRequest(status))
+      await expect(func).rejects.toThrowRedirectTo(CONTACT_SUMMARY.uri)
+    })
+
+    it('payment required returns false if price = 0', async () => {
+      const status = () => ({
+        [LICENCE_SUMMARY.page]: true,
+        [CONTACT_SUMMARY.page]: true
+      })
+      const transaction = () => ({
+        permissions: [
+          {
+            licensee: {
+              firstName: 'Turanga',
+              lastName: 'Leela'
+            },
+            licenceType: 'trout-and-coarse',
+            numberOfRods: '2',
+            licenceLength: '8D',
+            permit: { cost: 0 }
+          },
+          {
+            licensee: {
+              firstName: 'Turanga',
+              lastName: 'Leela'
+            },
+            licenceType: 'trout-and-coarse',
+            numberOfRods: '2',
+            licenceLength: '8D',
+            permit: { cost: 0 }
+          }
+        ]
+      })
```

Comment:
Please can you make helper functions to create your transaction and status mocks, so that we don't end up with these large blocks in each test?